src/main/java/bio/terra/cloudevents/GCSEvent.java

```diff
@@ -2,6 +2,19 @@
 
 import java.util.Date;
 
+/**
+ * If we want to use {@link com.google.cloud.functions.BackgroundFunction} we need a class like this
+ * that contains the subset of data that we want to preserve from
+ * https://cloud.google.com/storage/docs/json_api/v1/objects#resource
+ *
+ * <p>In particular, the default Gson instance that GCFs use doesn't know how to deserialize to an
+ * {@link java.time.OffsetDateTime} so we have to change those fields to a date time representation
+ * that Gson can deserialize. Here we've chosen {@link Date} For this reason we can't use {@link
+ * com.google.events.cloud.storage.v1.StorageObjectData} in Background function which would be
+ * preferred.
+ *
+ * <p>https://cloud.google.com/functions/docs/writing/background
+ */
```

Comment:
Yeah, no reason at all I think Ivan just pulled this code from the sample and added some fields in case we need them.  If we go with `DeltaLayerRawFunction` we don't need this class at all which is my main reason for supporting that approach.

Reply:
Currently GCSEvent is still used by code that's in master so we can't delete just yet.